armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -680,10 +680,24 @@ def specifyTargetComponent(self, b, flagOfInterest=None):
         RuntimeError
             multiple target components found
         """
+
         if flagOfInterest is None:
-            componentWFlag = [c for c in b.getChildren() if c.p.flags in b.p.flags]
+            # Follow expansion of most neutronically important component, fuel first then control/poison
+            targetFlagsInPreferedOrder = [
+                Flags.FUEL,
+                Flags.CONTROL,
+                Flags.POISON,
+                Flags.SHIELD,
+            ]
+            for targetFlag in targetFlagsInPreferedOrder:
+                componentWFlag = [c for c in b.getChildren() if c.hasFlags(targetFlag)]
+                if componentWFlag != []:  # or just `if componentWFlag`
+                    break
         else:
             componentWFlag = [c for c in b.getChildren() if c.hasFlags(flagOfInterest)]
+        # some blocks/components are not included in the above list but should still be found
+        if not componentWFlag:
```

Comment:
this block should probably be after line 695. If the flag of interest isn't present this should still be an error.

Reply:
agreed, done